src/NUnitFramework/framework/Internal/Randomizer.cs

```diff
@@ -563,18 +563,67 @@ public string GetString()
         /// </summary>
         public decimal NextDecimal()
         {
-            int low = Next(0, int.MaxValue);
-            int mid = Next(0, int.MaxValue);
-            int high = Next(0, int.MaxValue);
-            return new Decimal(low, mid, high, false, 0);
+            var bytes = new byte[sizeof(int) * 3];
+            NextBytes(bytes);
+
+            var low = BitConverter.ToInt32(bytes, 0);
+            var mid = BitConverter.ToInt32(bytes, 4);
+            var high = BitConverter.ToInt32(bytes, 8);
+
+            return new decimal(low, mid, high, false, 0);
         }
 
         /// <summary>
         /// Returns a random decimal between positive zero and the specified maximum.
         /// </summary>
         public decimal NextDecimal(decimal max)
```

Comment:
I think we need to document the algorithm in a comment.

Reply:
Good idea.